### PR TITLE
KFSPTS-30711 Fix general ledger entries for stopped checks

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java [hunk 1]
```diff
@@ -175,9 +175,14 @@ private void populatePaymentGeneralLedgerPendingEntry(
             glPendingTransaction.setAccountNumber(paymentAccountDetail.getAccountNbr());
             glPendingTransaction.setSubAccountNumber(paymentAccountDetail.getSubAccountNbr());
             glPendingTransaction.setChartOfAccountsCode(paymentAccountDetail.getFinChartCode());
-
-            glPendingTransaction.setFinancialDocumentTypeCode(paymentAccountDetail.getPaymentDetail().getFinancialDocumentTypeCode());
-
+            
+            if (StringUtils.equals(paymentGroup.getDisbursementType().getCode(), PdpConstants.DisbursementTypeCodes.ACH)) {
+                glPendingTransaction.setFinancialDocumentTypeCode(state.documentTypeForDisbursementType(PdpConstants.DisbursementTypeCodes.ACH));
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java:180]: The previous set functions were
glPendingTransaction.setFinancialDocumentTypeCode(achFdocTypeCode);
or
 glPendingTransaction.setFinancialDocumentTypeCode(checkFdocTypeCod);

I am not sure if achFdocTypeCode or checkFdocTypeCod  exist any longer, but you could use the PDP constant.
Why do you need to reference the state object?

**Review by jhulslander — COMMENTED** [review]
I had a question, and I think the PR description needs to be updated so the commit message is meaningful

File: src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java (hunk 1 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java:180]: Good question! KualiCo added the GeneratePdpGlpeState object with the 1/4 upgrade and this has methods that will get the doc type for the given disbursement type. Different maps are used for the different cases like stop, stale cancelled, reissued, etc. This method was updated to follow base code changes.

Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java:180]: Thanks, I shall merge

**Review by jhulslander — APPROVED** [review]
my question was answered, the unit tests pass, I shall merge

**Pull request merged**